### PR TITLE
Feature/misc

### PR DESCRIPTION
**Pull request by garyluu**
Misc issues
- Puts the publish, refresh, delete buttons in the button group for alignment
- Reordered the files displayed to ignore validity (added an invalid version alert which should be moved elsewhere later).  ga4gh/dockstore#1673
- Add validation to reflect the changes to the webservice (can't have a wdl test parameter file and cwl test parameter file with the same name)
- Workaround to cut the number of GA4GH files related calls by half (apparently the permissions end up changing which causes the change detection to trigger).

### CODE REVIEW

**Comment by garyluu** [comment]
Made a few changes since.  Handle the case where the unsaved cwl and wdl test files are identical, fix tests, fix the 'cwl' and 'wdl' vs DescriptorType.CWL and DescriptorType.WDL mixup

File: src/app/container/add-tag/add-tag.component.ts [hunk 1]
```diff
@@ -179,22 +179,20 @@ export class AddTagComponent implements OnInit, AfterViewChecked {
...
-      } else {
-        return false;
-      }
+  // TODO: This is also executed a bajillion times
+  hasDuplicateTestJson(type: ToolDescriptor.TypeEnum) {
+    if (type === this.DescriptorType.CWL) {
+      return this.unsavedWDLTestParameterFilePaths.includes(this.unsavedTestCWLFile) ||
+        this.unsavedCWLTestParameterFilePaths.includes(this.unsavedTestCWLFile) ||
+        (this.unsavedTestCWLFile === this.unsavedTestWDLFile && this.unsavedTestCWLFile);
+    } else if (type === this.DescriptorType.WDL) {
```
Comment (denis-yuen) [inline src/app/container/add-tag/add-tag.component.ts]: these predicates are quite complex, could use a comment at least
potential for simplification possibly (is this the same as merging unsavedWDLTestParameterFilePaths and unsavedWDLTestParameterFilePaths and checking whether the merge contains the unsaved test file? not sure what is going on in the last bracket 

so: 
minimum: code comment what each branch is doing and what is going on in the last bracket of each branch
stretch goal: simplify this

File: src/app/container/add-tag/add-tag.component.html [hunk 2]
```diff
@@ -97,7 +97,7 @@ <h4 class="modal-title">Add Version Tag</h4>
               <div class="input-group">
                 <input [ngClass]="{ 'input-right-button' : editMode }" type="text" #model1="ngModel" class="form-control" name="cwlTestParameterFilePath" [(ngModel)]="unsavedTestCWLFile" minlength="3" maxlength="128" [pattern]="validationPatterns.testFilePath" title="Relative path to a CWL Test Parameter File in the Git repository." placeholder="e.g. /test.cwl.json" [disabled]="!editMode" ng-class="editMode ? 'test_parameter_input' : ''"/>
                 <span class="input-group-btn">
-                  <button title="Add CWL test parameter file" type="button" class="btn btn-default form-sm-button" [disabled]="hasDuplicateTestJson('cwl')" (click)="addTestParameterFile(DescriptorType.CWL)" *ngIf="editMode && !(model1.invalid || unsavedTestCWLFile.length == 0)">
+                  <button title="Add CWL test parameter file" type="button" class="btn btn-default form-sm-button" [disabled]="hasDuplicateTestJson(DescriptorType.CWL)" (click)="addTestParameterFile(DescriptorType.CWL)" *ngIf="editMode && !(model1.invalid || unsavedTestCWLFile.length == 0)">
```
Comment (denis-yuen) [inline src/app/container/add-tag/add-tag.component.html]: :+1:

File: src/app/container/version-modal/version-modal.component.ts [hunk 3]
```diff
@@ -277,20 +277,19 @@ export class VersionModalComponent implements OnInit, AfterViewChecked {
...
+  // TODO: This code is repeated in add-tag.component.ts for tools, move it somewhere common
+  // TODO: This is also executed a bajillion times
+  hasDuplicateTestJson(type: ToolDescriptor.TypeEnum) {
     if (type === this.DescriptorType.CWL) {
-      if (this.unsavedCWLTestParameterFilePaths.includes(this.unsavedTestCWLFile)) {
-        return true;
-      } else {
-        return false;
-      }
+      return this.unsavedWDLTestParameterFilePaths.includes(this.unsavedTestCWLFile) ||
```
Comment (denis-yuen) [inline src/app/container/version-modal/version-modal.component.ts]: same deal, except in this case, might be possible to share a utility function
looks like a copy and paste

File: src/app/loginComponents/accounts/external/accounts.component.html [hunk 4]
```diff
@@ -43,6 +43,9 @@ <h4 class="panel-title">{{ row.name }} Account
         </div>
         <div class="panel-body">
           <strong>{{ row.bold }} </strong> {{ row.message }}
+          <p *ngIf="row.name == 'Google'">
+            "Login with Google" will prioritize logging into an existing account that has a Google token
```
Comment (denis-yuen) [inline src/app/loginComponents/accounts/external/accounts.component.html]: can revert this based on yesterday discussion

File: src/app/container/version-modal/version-modal.component.ts (hunk 3 above)
Comment (garyluu) [reply src/app/container/version-modal/version-modal.component.ts]: ya, it's in the TODO already.  didn't want to keep stretching this misc PR any further

Comment (denis-yuen) [reply src/app/container/version-modal/version-modal.component.ts]: sounds good, then create an issue to remember and then move on

**Pull request merged**